event-sourcing/event-subscription/src/main/java/uk/gov/justice/subscription/EventSourcesParser.java

```diff
@@ -36,7 +36,10 @@ public EventSourcesParser(final YamlParser yamlParser, final YamlFileValidator y
      * @param urls the YAML URLs to parse
      * @return Stream of {@link EventSourceDefinition}
      */
-    public Stream<EventSourceDefinition> eventSourcesFrom(final Collection<URL> urls) {
+    public Stream<EventSourceDefinition> eventSourcesFrom(final List<URL> urls) {
+        if ( urls == null || urls.isEmpty()){
```

Comment:
I don't think this check is in the correct place...the check should happen in the calling class, as that class knows from which location it has retrieved the event-source.yaml URLs. I would also probably not null check unless you know a null is possible.